lsp-types/src/Language/LSP/Types/SMethodMap.hs

```diff
@@ -20,25 +20,34 @@ import qualified Data.IntMap.Strict as IntMap
 import Data.Kind (Type)
 import Data.Map (Map)
 import qualified Data.Map.Strict as Map
+import Data.Some
 import Data.Text (Text)
 import GHC.Exts (Int(..), dataToTag#)
 import Unsafe.Coerce (unsafeCoerce)
 
 import Language.LSP.Types.Method (Method(..), SMethod(..))
 
+-- This type exists to avoid a dependency on 'dependent-map'. It is less
+-- safe (since we use 'unsafeCoerce') but much simpler and hence easier to include.
+-- | A specialized altenative to a full dependent map for use with 'SMethod'.
 data SMethodMap (v :: Method f t -> Type) =
-  SMethodMap !(IntMap (v 'CustomMethod)) !(Map Text (v 'CustomMethod))
+  -- This works by using an 'IntMap' indexed by constructor tag for the majority
+  -- of 'SMethod's, which have no parameters, and hence can only appear once as keys
+  -- in the map.
+  -- The sole exception is 'SCustomMethod', for which we keep a separate map from
+  -- its 'Text' parameter.
+  SMethodMap !(IntMap (Some v)) !(Map Text (v 'CustomMethod))
```

Comment:
`Some` is deeply magical and spiced with a generous amount of `unsafeCoerce` under the hood, which apparently breaks something in tests. If you wish to underline that the index of `v` is irrelevant, I'd suggest writing `IntMap (v GHC.Exts.Any)` - no other changes are required.

Reply:
I'm not sure I understand why this is failing (tests pass locally for me :thinking:), but I agree that your suggestion achieves the same thing. I'll return to that.